runtimes/runtimes/standalone.ts

```diff
@@ -181,6 +182,61 @@ export const standalone = (props: RuntimeProps) => {
                         })
                     }),
                 getFileSize: path => stat(path),
+                getServerDataFolder: serverName => {
+                    const initializeParams = lspRouter.clientInitializeParams
+                    const clientSpecifiedLocation = initializeParams?.initializationOptions?.aws.clientDataFolder
+                    if (clientSpecifiedLocation) {
+                        return path.join(clientSpecifiedLocation, serverName)
+                    }
+
+                    const clientFolderName = getClientNameFromParams(initializeParams)
+                    const standardizedClientFolderName = standardizeFileName(clientFolderName)
+
+                    const appDataFolder = getPlatformAppDataFolder()
+                    return appDataFolder
+                        ? path.join(appDataFolder, standardizedClientFolderName, serverName)
+                        : path.join(os.homedir(), `.${standardizedClientFolderName}`, serverName)
+
+                    function getPlatformAppDataFolder(): string {
+                        const platform = process.platform
+                        switch (platform) {
+                            case 'win32':
+                                return process.env.APPDATA || path.join(os.homedir(), 'AppData', 'Roaming')
+
+                            case 'darwin':
+                                return path.join(os.homedir(), 'Library', 'Application Support')
+
+                            case 'linux':
+                                return process.env.XDG_DATA_HOME || path.join(os.homedir(), '.local', 'share')
+
+                            default:
+                                return ''
```

Comment:
Should the default here just be os.homedir()?  Seems safer than calling code, if not checking the result, blindly adding a relative path to an empty string and attempting to use it.

Reply:
Yes I was thinking about that. The default case is different than the rest because we want to append a hidden directory to it. But I agree it is more coherent to return `os.homedir()` and decide how we append the folder name in the caller function, I'll adjust it.